src/main/java/uk/gov/nationalarchives/pdi/step/jena/groupmerge/JenaGroupMergeStepMeta.java

```diff
@@ -134,87 +136,92 @@ public String getXML() throws KettleException {
-                        final Node jenaModelFieldNode = jenaModelFieldNodes.get(i);
-
-                        final String xFieldName = XMLHandler.getTagValue(jenaModelFieldNode, ELEM_NAME_FIELD_NAME);
-                        if (isNullOrEmpty(xFieldName)) {
-                            continue;
-                        }
-
-                        final String xActionIfNoSuchField = XMLHandler.getTagValue(jenaModelFieldNode, ELEM_NAME_ACTION_IF_NO_SUCH_FIELD);
-                        final ActionIfNoSuchField actionIfNoSuchField = isNotEmpty(xActionIfNoSuchField) ? ActionIfNoSuchField.valueOf(xActionIfNoSuchField) : ActionIfNoSuchField.ERROR;
-                        final String xActionIfNull = XMLHandler.getTagValue(jenaModelFieldNode, ELEM_NAME_ACTION_IF_NULL);
-                        final ActionIfNull actionIfNull = isNotEmpty(xActionIfNull) ? ActionIfNull.valueOf(xActionIfNull) : ActionIfNull.ERROR;
-                        this.jenaModelFields.add(new ConstrainedField(xFieldName, actionIfNoSuchField, actionIfNull));
+                this.mergeFields = new ArrayList<>();
+
+                for (final Node jenaModelFieldNode : jenaModelFieldNodes) {
+                    final String xFieldName = XMLHandler.getTagValue(jenaModelFieldNode, ELEM_NAME_FIELD_NAME);
+                    if (isNullOrEmpty(xFieldName)) {
+                        continue;
                     }
+
+                    final String xActionIfNoSuchField = XMLHandler.getTagValue(jenaModelFieldNode, ELEM_NAME_ACTION_IF_NO_SUCH_FIELD);
+                    final ActionIfNoSuchField actionIfNoSuchField = isNotEmpty(xActionIfNoSuchField) ? ActionIfNoSuchField.valueOf(xActionIfNoSuchField) : ActionIfNoSuchField.ERROR;
+                    final String xActionIfNull = XMLHandler.getTagValue(jenaModelFieldNode, ELEM_NAME_ACTION_IF_NULL);
+                    final ActionIfNull actionIfNull = isNotEmpty(xActionIfNull) ? ActionIfNull.valueOf(xActionIfNull) : ActionIfNull.ERROR;
+                    final String xMutateFirstModel = XMLHandler.getTagValue(jenaModelFieldNode, ELEM_NAME_MUTATE_FIRST_MODEL);
+                    final MutateFirstModel mutateFirstModel = isNotEmpty(xMutateFirstModel) ? MutateFirstModel.valueOf(xMutateFirstModel) : MutateFirstModel.YES;
+                    final String xTargetFieldName = XMLHandler.getTagValue(jenaModelFieldNode, ELEM_NAME_TARGET_FIELD_NAME);
+                    final String targetFieldName = isNotEmpty(xTargetFieldName) ? xTargetFieldName : null;
+
+                    this.mergeFields.add(new ModelMergeConstrainedField(xFieldName, actionIfNoSuchField, actionIfNull, mutateFirstModel, targetFieldName));
                 }
             }
         }
+
+        final String xOtherFieldAction = XMLHandler.getTagValue(stepnode, ELEM_NAME_OTHER_FIELD_ACTION);
+        if (isNotEmpty(xOtherFieldAction)) {
+            this.otherFieldAction = OtherFieldAction.valueOf(xOtherFieldAction);
+        }
```

Comment:
Do we not need to set a default value if none is found in the XML file?

Reply:
Yes, I think that actually does happen already here - https://github.com/nationalarchives/kettle-jena-plugins/pull/51/files/bf3cc82bf4d3dcd318111c71a80f75e028892906#diff-033185489e0fe3db1ed2ce623eaef295d535d06f588ce81b3fa020bc7f559b48R106

Reply:
That link doesn't work - however I expect you are pointing to the `setDefault()` method but I don't think this will get called unless you are creating a new step or editing an existing one. If there already happens to be a step in the workflow without this `otherFieldAction` set you will get a null pointer if you try to save the workflow.
I think the solution is to wrap the call like so..
```
builder.append(XMLHandler.addTagValue(ELEM_NAME_OTHER_FIELD_ACTION, emptyIfNull(otherFieldAction.name())));
````

Reply:
I just pushed an additional commit that hopefully solves it now